src/libraries/illink-oob.targets

```diff
@@ -1,63 +1,78 @@
-      <_OOBReferences Include="@(_OOBReferencesFileName -> '%(OriginalIdentity)')" />
-      <_OOBReferences Include="$(SystemPrivateCoreLibPath)" />
+      <OOBAssemblyToIgnore Include="System.CodeDom" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition" />
+      <OOBAssemblyToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <OOBAssemblyToIgnore Include="System.Composition.AttributedModel" />
+      <OOBAssemblyToIgnore Include="System.Composition.Convention" />
+      <OOBAssemblyToIgnore Include="System.Composition.Hosting" />
+      <OOBAssemblyToIgnore Include="System.Composition.Runtime" />
+      <OOBAssemblyToIgnore Include="System.Composition.TypedParts" />
+      <OOBAssemblyToIgnore Include="System.Configuration.ConfigurationManager" />
+      <OOBAssemblyToIgnore Include="System.Speech" />
+
+      <NetCoreAppCurrentAssembly Include="$(NetCoreAppCurrentRuntimePath)*.dll"
+                                 Exclude="$(NetCoreAppCurrentRuntimePath)*.Generator.dll;
+                                          $(NetCoreAppCurrentRuntimePath)*.Native.dll;
+                                          $(NetCoreAppCurrentRuntimePath)*msquic.dll" />
+      <SharedFrameworkAssembly Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+
+      <!-- Move items to FileName so that we can subtract them -->
+      <NetCoreAppCurrentAssemblyWithFilename Include="@(NetCoreAppCurrentAssembly -> '%(FileName)')"
+                                             OriginalIdentity="%(Identity)" />
+      <SharedFrameworkAssemblyWithFilename Include="@(SharedFrameworkAssembly -> '%(FileName)')"
+                                           OriginalIdentity="%(Identity)" />
+
+      <OOBAssemblyToTrimWithFileName Include="@(NetCoreAppCurrentAssemblyWithFilename)"
+                                     Exclude="@(SharedFrameworkAssemblyWithFilename);
+                                              @(OOBAssemblyToIgnore)" />
+      <OOBAssemblyToTrim Include="@(OOBAssemblyToTrimWithFileName -> '%(OriginalIdentity)')" />
+      
+      <OOBAssemblyReferenceWithFilename Include="@(NetCoreAppCurrentAssemblyWithFilename)"
+                                        Exclude="@(OOBAssemblyToTrimWithFileName)" />
+      <OOBAssemblyReference Include="@(OOBAssemblyReferenceWithFilename -> '%(OriginalIdentity)')" />
+      <OOBAssemblyReference Include="$(SystemPrivateCoreLibPath)" />
     </ItemGroup>
 
     <ItemGroup>
-      <!-- Include suppression XML files bin-placed in earlier per-library linker run. -->
-      <_OOBSuppressionsXmls Include="$(ILLinkTrimAssemblyOOBSuppressionsXmlsDir)*.xml" />
+      <OOBLibraryTrimmed Include="$(OOBAssembliesTrimmedArtifactsPath)*.*" />
     </ItemGroup>
 
+  </Target>
+
+  <Target Name="ILLinkTrimOOBAssemblies"
+          AfterTargets="Build"
+          DependsOnTargets="GetOOBAssembliesToTrim;PrepareForAssembliesTrim"
```

Comment:
NIT: Technically you don't need to also depend on PrepareForAssembliesTrim, since this target depends on GetOOBAssembliesToTrim, and that one in turn depends on PrepareForAssembliesTrim.

Reply:
I prefer targets to list their direct dependencies even if some might come transitively via other targets.